code/modules/mining/equipment/kheiral_cuffs.dm

```diff
@@ -0,0 +1,120 @@
+	desc = "A prototype wrist communicator powered by Kheiral Matter. When both ends are clamped to one wrist, can send a distress signal upon your death off-station.\nA small engraving on the inside reads, \"NOT HANDCUFFS\""
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "strand"
+	worn_icon_state = "strandcuff"
+	slot_flags = ITEM_SLOT_GLOVES
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	gender = PLURAL
+	throw_speed = 3
+	throw_range = 5
+	/// If we're in the glove slot
+	var/on_wrist = FALSE
+	/// If the GPS is already on
+	var/gps_enabled = FALSE
+	/// If we're off the station's Z-level
+	var/far_from_home = FALSE
+	/// Used to broadcast user's death through the radio
+	var/obj/item/radio/internal_radio
+
+/obj/item/kheiral_cuffs/Initialize()
+	. = ..()
+	update_icon(UPDATE_OVERLAYS)
+	RegisterSignal(src, COMSIG_MOVABLE_Z_CHANGED, .proc/check_z)
+	internal_radio = new/obj/item/radio(src)
+	internal_radio.subspace_transmission = TRUE
+	internal_radio.canhear_range = 0
+	internal_radio.should_be_listening = FALSE
+	internal_radio.keyslot = new /obj/item/encryptionkey/strandcuffs
+	internal_radio.set_listening(FALSE)
+	internal_radio.recalculateChannels()
+
+/obj/item/kheiral_cuffs/examine(mob/user)
+	. = ..()
+	if(gps_enabled)
+		. += span_notice("The cuff's GPS signal is on.")
+
+/obj/item/kheiral_cuffs/item_action_slot_check(slot)
+	return slot == ITEM_SLOT_GLOVES
+
+/obj/item/kheiral_cuffs/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot != ITEM_SLOT_GLOVES)
+		return
+	on_wrist = TRUE
+	playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	connect_kheiral_network(user)
+
+/obj/item/kheiral_cuffs/dropped(mob/user, silent)
+	. = ..()
+	on_wrist = FALSE
+	playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	remove_kheiral_network(user)
+
+/// Enables the GPS & registers the statchange signal
+/obj/item/kheiral_cuffs/proc/connect_kheiral_network(mob/user)
+	if(gps_enabled)
+		return
+	if(on_wrist && far_from_home)
+		AddComponent(/datum/component/gps, "*[user.name]'s Kheiral Link")
+		balloon_alert(user, "GPS activated")
+		RegisterSignal(user, COMSIG_MOB_STATCHANGE, .proc/on_user_statchange)
+		gps_enabled = TRUE
+
+/// Disables the GPS and unregisters the statchange signal
+/obj/item/kheiral_cuffs/proc/remove_kheiral_network(mob/user)
+	if(!gps_enabled)
+		return
+	if(!on_wrist || !far_from_home)
+		qdel(GetComponent(/datum/component/gps))
+		balloon_alert(user, "GPS de-activated")
+		UnregisterSignal(user, COMSIG_MOB_STATCHANGE)
+		gps_enabled = FALSE
+
+/obj/item/kheiral_cuffs/Destroy(force)
+	. = ..()
+	UnregisterSignal(src, COMSIG_MOVABLE_Z_CHANGED)
+	if(internal_radio)
+		qdel(internal_radio)
+
+/// If we're off the Z-level, set far_from_home = TRUE. If being worn, trigger kheiral_network proc
+/obj/item/kheiral_cuffs/proc/check_z(datum/source, turf/old_turf, turf/new_turf)
+	SIGNAL_HANDLER
+
+	if(!isturf(new_turf))
+		return
+
+	var/mob/living/bridges = loc
```

Comment:
Is this always going to be on mob/living? If it can reach here while inside of a storage object or crate, it should probably check before being assigned.

Reply:
You're right that I should put the assignment after the check, though in my testing it didn't really seem to matter since the `bridges` var is only used to check if it's already being worn so it can trigger the GPS immediately upon changing Z-levels instead of requiring the user to take them off and put them back on.

Reply:
Actually no, I can't put the assignment after, because it's used in two different cases.

I might be misunderstanding your question as well, to be completely honest.